### DISCUSSION TITLE
Estimating a non-linear model

### DISCUSSION THREAD

**Discussion by Nic2020** [IRIS-Solutions-Team/IRIS-Toolbox#317] [Q&A]
Hi. I am wondering what would be the best strategy to estimate a non-linear model. I put in the attached an honest attempt for an overly simplistic example with IRIS 20151016 and IRIS 20220222. Thank you for any insights on this.
[estimate_nonlinear.zip](https://github.com/IRIS-Solutions-Team/IRIS-Toolbox/files/8683632/estimate_nonlinear.zip)

**Comment by jaromir-benes** [comment]
Hi Nicholas - that was the risk of telling me not to do it on the weekend :))

I'm attaching a commented file.

There are two major things (apart from some clean up comments)

1. There was a bug in Iris. Please use the latest commit from the "bleeding" branch. I strongly recommend cloning the Iris GitHub repository; that way, you can very easily keep everything up to date and switch between versions.

2. When you're estimating parameters that alter the steady state of the model, you need to expressly request recalculation of the steady state in each iteration. By default this is not the case (simply because it's a very time consuming task) assuming that you only estimate parameters that affect the dynamics only. It's the responsibility of the modeler to activate this option when needed.

Let me know.

Best
Jaromir

[test_bleeding.m.txt](https://github.com/IRIS-Solutions-Team/IRIS-Toolbox/files/8715353/test_bleeding.m.txt)
